pkg/proxy/netstat.go

```diff
@@ -0,0 +1,52 @@
+package proxy
+
+import (
+	"fmt"
+	"io/ioutil"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	// procNetTCPFiles is the constant list of /proc/net files to read to get
+	// the same information about open TCP connections as output by netstat.
+	procNetTCPFiles = []string{
+		"/proc/net/tcp",
+		"/proc/net/tcp6",
+	}
+
+	// procNetTCPRegexp matches the first two columns of /proc/net/tcp* files
+	// and submatches on the local port number.
+	procNetTCPRegexp = regexp.MustCompile("^ *[[:digit:]]*: *[[:xdigit:]]*:([[:xdigit:]]*) ")
+)
+
+// readOpenLocalTCPPorts returns the set of TCP ports currently open locally.
+func readOpenLocalTCPPorts() (map[uint16]struct{}, error) {
+	openLocalPorts := make(map[uint16]struct{}, 128)
```

Comment:
Any particular reason for using 128?

Reply:
It's just very probably that the map will not be empty, so it's bad for sure to start with a map with capacity 0. I just randomly picked 128 as a random positive integer.